nixos/modules/services/backup/syncoid.nix

```diff
@@ -383,7 +345,23 @@ in
               RootDirectory = "/run/syncoid/${escapeUnitName name}";
               RootDirectoryStartOnly = true;
               BindPaths = [ "/dev/zfs" ];
-              BindReadOnlyPaths = [ builtins.storeDir "/etc" "/run" "/bin/sh" ];
+              BindReadOnlyPaths = [ builtins.storeDir "/etc" "/run" "/bin/sh"
+                # A custom LD_LIBRARY_PATH is needed to access in `getent passwd`
+                # the systemd's entry about the DynamicUser=,
+                # so that ssh won't fail with: "No user exists for uid $UID".
+                # Unfortunately, Bash is incompatible with libnss_systemd.so:
+                # https://www.mail-archive.com/bug-bash@gnu.org/msg24306.html
+                # Hence the wrapping of ssh is done here as a mounted path,
+                # because Nixpkgs' wrapping of syncoid enforces the use
+                # of the ${pkgs.openssh}/bin/ssh path.
+                # This problem does not arise on NixOS systems where stdenv.hostPlatform.libc == "musl",
+                # because then Bash is built with --without-bash-malloc
+                ("${pkgs.writeShellScript "ssh-with-support-for-DynamicUser" ''
```

Comment:
Is this still necessary now that the bash patches have been merged?

Reply:
Indeed it's no longer needed, I've removed it. Thanks.